### PR TITLE
Authorization with jwt

### PR DESCRIPTION
**Pull request by aaronshiel**
JWT is used for authorization:
- authorize_to_edit_mentor decorator validates that the requester is either editing one of their own mentors or has the permission level to edit other mentors
- authorize_to_manage_content decorate checks that the user is permitted to manage content (unrelated to other mentors, such as managing files on the server)

Paired with: https://github.com/mentorpal/mentor-graphql/pull/74
Paired with: https://github.com/mentorpal/mentor-admin/pull/159

### CODE REVIEW

File: mentor_upload_api/src/mentor_upload_api/authorization_decorator.py [hunk 1]
```diff
@@ -7,68 +7,107 @@
...
+from mentor_upload_api.helpers import validate_json
 
 log = logging.getLogger("authorization")
 
 
-class GQLQueryBody(TypedDict):
-    query: str
+jwt_payload_schema = {
+    "type": "object",
+    "properties": {
```
Comment (mbsimonovic) [inline mentor_upload_api/src/mentor_upload_api/authorization_decorator.py:21]: what happens if jwt contains additional properties? think there was a setting to ignore these

File: mentor_upload_api/src/mentor_upload_api/authorization_decorator.py [hunk 2]
```diff
@@ -7,68 +7,107 @@
...
+            json_body = json.loads(body)
+        else:
+            json_body = request.json
+        if not json_body:
+            raise Exception("missing required param body")
+
+        validate_json(json_body, authorize_edit_mentor_payload_schema)
+        mentor_being_edited = json_body["mentor"]
+
+        # Auth/authz requester via JWT data
```
Comment (mbsimonovic) [inline mentor_upload_api/src/mentor_upload_api/authorization_decorator.py:85]: can you somehow reuse this jwt parsing code?

File: mentor_upload_api/src/mentor_upload_api/authorization_decorator.py (hunk 1 above)
Comment (aaronshiel) [reply mentor_upload_api/src/mentor_upload_api/authorization_decorator.py:21]: additional properties are [allowed by default](http://json-schema.org/understanding-json-schema/reference/object.html#additional-properties)

**Review by mbsimonovic — APPROVED** [review]
[nice](https://www.youtube.com/watch?v=LBduNcf1eQc) 👏🏻

**Pull request merged**